Refactor libraryCardsEnabled method to AccountCapabilities class.

demiankatz opened the pull request:
This is a first step toward decoupling library cards enabled checks from user row objects. We need to do this in order to begin moving more logic into the UserCardService, which in turn is needed to complete #3625.

I don't think there's any way to do this that isn't somewhat messy. I welcome suggestions for better approaches.

TODO:
- [x] Merge #3631 and #3632 before this
- [x] Merge #3636 before this
- [x] Merge #3638 before this
- [x] Update changelog when merging (changes to \VuFind\Config\AccountCapabilities constructor/auth property; changes to \VuFind\Db\Row\User constructor)

demiankatz committed:
Refactor libraryCardsEnabled method to AccountCapabilities class.

demiankatz committed:
Add more UserEntityInterface methods.

demiankatz committed:
Refactor AuthManager to use service.

demiankatz committed:
Cleaner solution to circular dependencies.

demiankatz committed:
Use UserEntityInterface/UserService in LoginTokenManager.

demiankatz committed:
Minor improvement.

demiankatz committed:
Minor improvements.

demiankatz committed:
Fix typo.

demiankatz committed:
Merge branch 'userentityinterface-logintoken' into refactor-card-check

demiankatz committed:
Move circular dependency removal elsewhere.

demiankatz committed:
Merge branch 'dev' into refactor-card-check

demiankatz committed:
Merge branch 'dev' into refactor-card-check

demiankatz committed:
Progress.

demiankatz committed:
Fixes.

demiankatz committed:
Fix error.

demiankatz committed:
Merge branch 'dev' into refactor-card-check

demiankatz committed:
Fix typo.

demiankatz committed:
Improve comment.

demiankatz committed:
Use UserEntityInterface in NotifyCommand.

demiankatz committed:
Use UserEntityInterface in SwitchDbHasHCommand.

demiankatz committed:
Require AccountCapabilities in User object.

demiankatz committed:
phpcbf

demiankatz committed:
Merge branch 'dev' into refactor-card-check

demiankatz committed:
Roll back unnecessary changes.

demiankatz commented:
I think this is now ready for review. As discussed elsewhere (probably on Slack) it might make sense to move the `inPrivacyMode` method into AccountCapabilities as well... but this will only strengthen the existing circular dependencies, so I'm not sure if it's a good idea. Either way, I think that's a separate matter that can be handled outside of this current PR.

EreMaijala reviewed (approved):
Based on the description I was expecting something much more messy than this. Looks good to me!

demiankatz commented:
> Based on the description I was expecting something much more messy than this. Looks good to me!

It started out a lot messier before I split big chunks of it off into separate PRs and simplified everything. :-)
